LPS-130360 Migrate selection modals in commerce-channel-web

markocikos opened the pull request:
JIRA: https://issues.liferay.com/browse/LPS-130360

@jonmak08 @kresimir-coko 

I'm opening this as a draft for a quick review, before forwarding to commerce team: 
- I opted to change how data is represented in the search container, form `dataset` on row to `dataset` on button. With this, it fits our standard case for single selection, so we don't need custom events firing. This is a bit controversial, as it is not something we should normally do. Normally, we should leave data representation as it is and see if our utility handles it. I chose to it in this case because it was simple to do. It's on borderline of what the migration epic should do.
- Our frontend-js-web `delegate` utility has a pretty odd usage in JSPs, with `<aui:script require`. Because I needed the `liferay-search-container` module, I did
```js
<aui:script require="frontend-js-web/liferay/delegate/delegate.es as delegateModule">
	AUI().use('liferay-search-container', () => {
        ...
```
I am not sure if there is a better way. I would expect `delegate` to be exposed as `Liferay.Util.delegate`, is there a reason why we are not doing it that way?


Test case:
1. Have a site created using Minium template. This is a prerequisite for all commerce tasks:
   1.   Contol Panel > Sites > (+) Minium
   1.   Contol Panel > Search > Index Actions > Reindex all search indexes.
1. Create a blank site, Contol Panel > Sites > Blank Site
1. Commerce > Channels > open a channel > Type > Select Site. Save.
1. Delete a site. Save.

![after](https://user-images.githubusercontent.com/5435511/116121630-19ff6900-a6c1-11eb-89dc-c21a2ab9d0e7.gif)

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-18.liferay.com/job/test-portal-source-format/5586/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-18.liferay.com//userContent/jobs/test-portal-source-format/builds/5586/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest(master)/5735/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-18.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/c4c731285388d615b0631d9b2d7ead8b2555c486">c4c731285388d615b0631d9b2d7ead8b2555c486</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPS-130360">LPS-130360</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/3b36394806da77bf994cac3e4491eb3f19f6bd2b">3b36394806da77bf994cac3e4491eb3f19f6bd2b</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-18.liferay.com/job/test-portal-source-format/5586/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-18.liferay.com/userContent/jobs/test-portal-source-format/builds/5586/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-18.liferay.com/job/test-portal-source-format/5586/">test-portal-source-format#5586</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-18.liferay.com/userContent/jobs/test-portal-source-format/builds/5586/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1002">liferay-frontend#1002</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1528598769">[master] ci:test:sf - markocikos &gt; liferay-frontend - PR#1002 - 2021-04-26[09:57:59]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-18.liferay.com/job/publish-testray-report/493/">publish-testray-report#493</a><br/></div></div>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 9 out of 9 jobs passed</h3><h3>:x: ci:test:relevant - 20 out of 24 jobs passed in 5 hours 7 minutes</h3><details><summary>Click here for more details.</summary><br/><p>This pull is eligible for <a href="https://grow.liferay.com/share/CI+liferay-continuous-integration+GitHub+Commands#General-Commands">reevaluation</a>. When this <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)/1798">upstream build</a> has completed, using the following CI command will compare this pull request result against a more recent upstream result:<pre>ci:reevaluate:1187401_5735</pre></p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/c4c731285388d615b0631d9b2d7ead8b2555c486">c4c731285388d615b0631d9b2d7ead8b2555c486</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/053984004354b73692dbc66c41ca271ab9cba057">053984004354b73692dbc66c41ca271ab9cba057</a><br/>Jenkins Build URL: <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)//1797">Acceptance Upstream DXP (master) #1797</a></p><details><summary><strong>ci:test:stable - 9 out of 9 jobs PASSED</strong></summary><details><summary><strong>9 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432331/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432153/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432444/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328389/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469573/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328390/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469575/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469572/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469576/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 20 out of 24 jobs PASSED</strong></summary><div><h4>4 Failed Jobs:</h4><ul><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest(master)/5735/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469768/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469577/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/441154/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</strong></strike></a></li></ul></div><details><summary><strong>20 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/460696/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432442/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432331/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432335/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/8280/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-20.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432153/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432444/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/460697/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/460695/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469574/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328389/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432334/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432443/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469573/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432333/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/328390/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469575/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469572/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/432332/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469576/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><h5>For more details click <a href="https://test-1-18.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469577/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>1 Test Passed.<br/>0 Tests Failed.</p></div><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=2,label_exp=!master/469577//consoleText">AXIS_VARIABLE=2,label_exp=!master #469577</a><pre><code>Build was aborted</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=3,label_exp=!master/469577//consoleText">AXIS_VARIABLE=3,label_exp=!master #469577</a><pre><code>Build was aborted</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=4,label_exp=!master/469577//consoleText">AXIS_VARIABLE=4,label_exp=!master #469577</a><pre><code>Build was aborted</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//consoleText">AXIS_VARIABLE=1,label_exp=!master #469577</a><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testCompletePayment">com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testCompletePayment</a><pre><code>java.lang.AssertionError: 
[
SCENARIO When a payment is completed the payment status should be 'paid'
GIVEN An order with valid products
WHEN I complete the order payment
THEN The order payment status should be 'paid'
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testProcessPayment">com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testProcessPayment</a><pre><code>java.lang.AssertionError: 
[
SCENARIO When a payment is started the payment status should be 'authorized'
GIVEN An order with valid products
WHEN I start the order payment
THEN The order payment status should be 'authorized'
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logg...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.price.list.pricing.test/CommercePriceListLowestDiscoveryTest/testRetrieveCorrectPriceListByLowestEntry">com.liferay.commerce.price.list.pricing.test.CommercePriceListLowestDiscoveryTest.testRetrieveCorrectPriceListByLowestEntry</a><pre><code>java.lang.AssertionError: 
[
SCENARIO When multiple price list are defined for the same catalog the price list that provides the lowest price entry shall be taken
GIVEN A catalog with multiple price lists and one product
WHEN The price list is discovered
THEN The price list that gives the lowest price is retrieved
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.a...</code></pre></div></li><li>...</li></ol></div></li></ol><strong>Click <a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469577/testReport">here</a> for more failures.</strong></div></li><li><div><h5><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/441154/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/441154//consoleText">AXIS_VARIABLE=0,label_exp=!master #441154</a><pre><code>Triggering projects: verification-node
[liferay-jenkins-ee] $ /bin/bash -e -x /tmp/jenkins2380674242536347895.sh
+ echo 'Update the workspace repository for post-build steps.'
Update the workspace repository for post-build steps.
+ git clean -d -f -x
Removing git/
Removing hs_err_pid20744.log
+ git reset --hard
HEAD is now at 49f4836855 LRCI-2190 Remove all unused testray properties from build.properties
+ set +e
+ git show-ref --verify --quiet refs/heads/master-temp-1619462508683
+ VERIFY_BRANCH_RESULT=0
+ set -e
+ [[ 0 == 0 ]]
+ git checkout master-temp-1619462508683
Already on 'master-temp-1619462508683'
+ git log -n 1
commit 49f4836855420771e5501f017afd3242c81e77b2
Author: Michael Hashimoto &lt;michael.hashimoto@liferay.com&gt;
Date:   Thu Apr 22 11:50:08 2021 -0700

    LRCI-2190 Remove all unused testray properties from build.properties
+ echo 'Exit shell.'
Exit shell.
[liferay-jenkins-ee] $ /bin/bash -e -x /tmp/jenkins7934807779021899808.sh
+ echo 'Execute: build-test-portal-acceptance-pullrequest-batch.sh stop-current-job'
Execute: build-test-portal-acceptance-pullrequest-batch.sh stop-current-job
+ commands/build-test-portal-acceptance-pullrequest-batch.sh stop-current-job
Execute: ant -f commands/build-common.xml run-ant-target -Dant.file.name=build-test-portal-acceptance-pullrequest-batch.xml -Dant.target=stop-current-job -lib commands/../lib/jenkins
Java HotSpot(TM) 64-Bit Server VM warning: INFO: os::commit_memory(0x00000006c0000000, 805306368, 0) failed; error='Cannot allocate memory' (errno=12)
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 805306368 bytes for committing reserved memory.
# An error report file with more information is saved as:
# /opt/dev/projects/github/liferay-jenkins-ee/hs_err_pid20811.log
[EnvInject] - Injecting environment variables from a build step.
[EnvInject] - [ERROR] - The given properties file path '/opt/dev/projects/github/liferay-jenkins-ee/stop.properties' doesn't exist.
[EnvInject] - [ERROR] - Missing file path was resolved from pattern '${WORKSPACE}/stop.properties' .
[EnvInject] - [ERROR] - Problems occurs on injecting env vars as a build step: java.io.IOException: The given properties file path '/opt/dev/projects/github/liferay-jenkins-ee/stop.properties' doesn't exist.
[description-setter] Description set: ${BUILD_DESCRIPTION}
Build step 'Execute Scripts' marked build as failure
Finished: FAILURE</code></pre></div></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 053984004354b73692dbc66c41ca271ab9cba057:</strong></summary><ol><li><div><h5><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469577/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>1 Test Passed.<br/>0 Tests Failed.</p></div><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/469577//consoleText">AXIS_VARIABLE=0,label_exp=!master #469577</a><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/469577//testReport/com.liferay.commerce.account.service.test/CommerceAccountLocalServiceTest/testAddBusinessCommerceAccount">com.liferay.commerce.account.service.test.CommerceAccountLocalServiceTest.testAddBusinessCommerceAccount</a><pre><code>java.lang.AssertionError: 
[
SCENARIO Adding a new business Commerce Account
GIVEN A company
 AND A user as Account Administrator
WHEN The Commerce Account is created
THEN The list of accounts contains only one Commerce Account
 AND That Commerce Account matches the one created before.
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.conf...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/469577//testReport/com.liferay.commerce.account.service.test/CommerceAccountLocalServiceTest/testAddPersonalCommerceAccount">com.liferay.commerce.account.service.test.CommerceAccountLocalServiceTest.testAddPersonalCommerceAccount</a><pre><code>java.lang.AssertionError: 
[
SCENARIO Adding a new personal Commerce Account
GIVEN A company
WHEN The Commerce Account is created
THEN The list of accounts contains only one Commerce Account
 AND That Commerce Account matches the one created before.
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/469577//testReport/com.liferay.commerce.account.service.test/CommerceAccountLocalServiceTest/testBusinessOrganizationCommerceAccountsVisibility">com.liferay.commerce.account.service.test.CommerceAccountLocalServiceTest.testBusinessOrganizationCommerceAccountsVisibility</a><pre><code>java.lang.AssertionError: 
[
SCENARIO Adding new business Commerce Accounts
GIVEN A company
WHEN The Commerce Accounts are created
 AND Organizations are added to them
THEN Check the visibility of that accounts for all the organizations
]
[
Message: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
]
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org...</code></pre></div></li><li>...</li></ol></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//consoleText">AXIS_VARIABLE=1,label_exp=!master #469577</a><ol><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.inventory.service.mvcc.test/CommerceInventoryMVCCTest/testReplenishmentItemMVCC">com.liferay.commerce.inventory.service.mvcc.test.CommerceInventoryMVCCTest.testReplenishmentItemMVCC</a><pre><code>java.lang.AssertionError: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:456)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.inventory.service.mvcc.test/CommerceInventoryMVCCTest/testWarehouseItemMVCC">com.liferay.commerce.inventory.service.mvcc.test.CommerceInventoryMVCCTest.testWarehouseItemMVCC</a><pre><code>java.lang.AssertionError: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:456)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j...</code></pre></div></li><li><div><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1,label_exp=!master/469577//testReport/com.liferay.commerce.inventory.service.mvcc.test/CommerceInventoryMVCCTest/testWarehouseMVCC">com.liferay.commerce.inventory.service.mvcc.test.CommerceInventoryMVCCTest.testWarehouseMVCC</a><pre><code>java.lang.AssertionError: {level=ERROR, loggerName=com.liferay.commerce.product.internal.model.listener.CompanyModelListener, message=com.liferay.portal.kernel.exception.NoSuchGroupException: No Group exists with the primary key 0
	at com.liferay.portal.test.rule.LogAssertionTestRule$LogAppender.append(LogAssertionTestRule.java:330)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.logParent(LoggerConfig.java:531)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:500)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:456)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j...</code></pre></div></li><li>...</li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/469768/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>48 Tests Passed.<br/>3 Tests Failed.</p></div><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=23,label_exp=!master/469768//consoleText">AXIS_VARIABLE=23,label_exp=!master #469768</a><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=23,label_exp=!master/469768//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CommerceAccelerators_MiniumSiteFullFlowCheckingOutViaOrderDetails_">LocalFile.CommerceAccelerators#MiniumSiteFullFlowCheckingOutViaOrderDetails</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/23/LocalFile.CommerceAccelerators_MiniumSiteFullFlowCheckingOutViaOrderDetails/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/23/LocalFile.CommerceAccelerators_MiniumSiteFullFlowCheckingOutViaOrderDetails/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/23/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=24,label_exp=!master/469768//consoleText">AXIS_VARIABLE=24,label_exp=!master #469768</a><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=24,label_exp=!master/469768//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CommerceAccelerators_SpeedwellSiteFullFlow_">LocalFile.CommerceAccelerators#SpeedwellSiteFullFlow</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/24/LocalFile.CommerceAccelerators_SpeedwellSiteFullFlow/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/24/LocalFile.CommerceAccelerators_SpeedwellSiteFullFlow/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/24/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=6,label_exp=!master/469768//consoleText">AXIS_VARIABLE=6,label_exp=!master #469768</a><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=6,label_exp=!master/469768//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CPCommerceOrders_EditOrderCheckingOutViaOrderDetails_">LocalFile.CPCommerceOrders#EditOrderCheckingOutViaOrderDetails</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/6/LocalFile.CPCommerceOrders_EditOrderCheckingOutViaOrderDetails/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/6/LocalFile.CPCommerceOrders_EditOrderCheckingOutViaOrderDetails/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-04/test-1-18/test-portal-acceptance-pullrequest(master)/5735/functional-tomcat90-mysql57-jdk8/0/6/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li></ol></details></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-18/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-18/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-18/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest(master)/5735/">test-portal-acceptance-pullrequest(master)#5735</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-18.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5735/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1002">liferay-frontend#1002</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1528640714">[master] ci:test:relevant - markocikos &gt; liferay-frontend - PR#1002 - 2021-04-26[09:57:53]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-18.liferay.com/job/publish-testray-report/508/">publish-testray-report#508</a><br/></div></div>

wincent commented:
> I am not sure if there is a better way. I would expect `delegate` to be exposed as `Liferay.Util.delegate`, is there a reason why we are not doing it that way?

I don't have any specific recollection of discussion about that, but I do know that in general we have tried to avoid growing the `Liferay` object more than we have to. We'd much rather people `import` from [here](https://github.com/liferay/liferay-portal/blob/4b4ff53717645e7bf4c4b373b5cb5d218fb1207a/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js#L31). The logic is, in part, that if we make it too comfortable to keep writing JS in JSP, too comfortable to keep using `AUI`, then we'll never transform our frontend code into something that could reasonably be considered as uniformly "modern". So a bit of friction ("syntactic vinegar") is actually useful in that sense.

The other part of this is that global objects are gradually considered to be evil because they complicate analysis (eg. discovering the relationships in the system between providers and consumers, and how the actions of one consumer can affect the behavior of others etc). This is especially true in the case of `Liferay` because it's not even set up in a single location: it's written to and augmented from all manner of places across the codebase, and without an encyclopedic knowledge of how the entire DXP stack works, it's very difficult to figure out _where_ each property is defined, and _when_.

Having said that, if it is _really_ painful to not have it on `Liferay.Util`, it would not cause the world to end if we added it there. It's a judgment call that you can make (I don't feel strongly about it, even though I do agree with the arguments above).

markocikos commented:
> The logic is, in part, that if we make it too comfortable to keep writing JS in JSP, too comfortable to keep using `AUI`, then we'll never transform our frontend code into something that could reasonably be considered as uniformly "modern". So a bit of friction ("syntactic vinegar") is actually useful in that sense.

TIL "syntactic vinegar" is a thing :slightly_smiling_face: 

I do realize and agree with many downsides of `Liferay.Util`. It is definitely something we should be moving away from. The only consideration we should have is the migration cost. For example, if the only way to migrate is to rewrite module frontend from JSPs to React, I'd prefer `Liferay.Util`. The nudge from better utility code would we negligible compared to overall required effort for migration, so we would only make it harder on ourselves.

But, there is another way. More secret, and dark way. We could use `<liferay-frontend:component>` and put `delegate` there. Is this worth promoting as a better alternative for `delegate`, or even all `Liferay.Util.*` usages in JSPs?

wincent commented:
😂

> But, there is another way. More secret, and dark way.

![gollum-secret-way](https://user-images.githubusercontent.com/7074/116246446-4a4f1200-a76a-11eb-9d8e-7051916394a8.png)

> Is this worth promoting as a better alternative for `delegate`, or even all `Liferay.Util.*` usages in JSPs?

I don't really know... 🤷 😕

kresimir-coko commented:
@markocikos 
> I opted to change how data is represented in the search container, form dataset on row to dataset on button. 

Does this mean that you replaced the 
```
Liferay.Util.getOpener().Liferay.fire(
	'<%= HtmlUtil.escapeJS(itemSelectedEventName) %>',
	{data: {id: data.id, name: data.name}}
);
```

with `data='<%= HashMapBuilder.<String, Object>put("id", group.getGroupId()).put("name", group.getName(locale)).build() %>'` on the button?

Asking because I have some cases with custom events firing but they're not handled by the search container (as it was so far in my migration PRs) so I can't just delete them and get the preexisting `openSelectionModal` tied to it to work properly.

markocikos commented:
> Does this mean that you replaced ... with ...

@kresimir-coko  Yes. `openSelectionModal` works ootb if data is represented as [`dataset` on the triggering element](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L458-L469).

If a module has `.selector-button` elemets, but data representation different than `trigger.dataset`, then we are forced to make a change. We either
1. modify data representation, like I did in this PR, or 
2. change trigger class and leave existing data representation. That way listener in utility will not trigger.

markocikos commented:
@wincent I force-pushed, moved all JS to `<liferay-frontend:componen>`
- I didn't move only `delegate`, but also `openSelectionModal` handling. Only nice imports this way, no script tags in JSP.
- I had to add package.json. Source formatter made me prefix module name with `@liferay` :eyes: 
- I removed `UnicodeFormatter.toString()` for delete icon HTML. As far as I can tell it is unnecessary, and it is causing problems when rendering through  `<liferay-frontend:componen>`.

markocikos commented on modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/CommerceChannelSite.js:
```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
```
Self-review: two copyrights??

I made this file by copying from somewhere else, not paying attention to jsdoc. Searching for it now, we have 10 files in DXP with double copyrights :facepalm:

markocikos replied:
@wincent I opened a source-formatter bug for this: https://issues.liferay.com/browse/LPS-131310

wincent replied:
Great. Yeah, that'll happen whenever somebody copies from a `dxp/` directory to a non-`dxp/` directory, or vice versa. The ESLint plugin and rule (`notice/notice`) isn't super smart (we didn't write it — if we had, it would surely be 🧠), so all it does is a simple textual comparison: if it doesn't see the header it wants, it _adds_ it — it does not try to _replace_ an existing, wrong header.

If this happens a lot, we can try [the `onNonMatchingHeader` setting](https://www.npmjs.com/package/eslint-plugin-notice#onnonmatchingheader) (which I didn't use at first because it didn't seem super robust; that may have change by now), or submitting a feature request upstream, or we can just roll our own... I mean, we already have [24 in-house custom ESLint rules](https://github.com/liferay/liferay-frontend-projects/tree/master/projects/eslint-config), so we have plenty of examples (even just limiting it to our own rules) of how to do this.

wincent commented:
> I had to add package.json. Source formatter made me prefix module name with `@liferay` 👀

That's great. Working [as intended](https://github.com/liferay/liferay-frontend-projects/pull/500), so that we don't get p4wn3d by [a dependency confusion attack](https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/general/security/dependency_confusion.md).

The only reason more modules don't have `@liferay/` prefixes is because changing the name of existing modules is a bit harder than creating new ones (you have to update all the references when you change the name). In fact, we've only ever renamed two... and the rest are still waiting for us to get around to [LPS-129670](https://issues.liferay.com/browse/LPS-129670) — personally, not holding my breath for that one... 😂

Anyway, there are no external references to `commerce-channel-web`, so "renaming" it in this way won't break anything, so I think that's all good.

kresimir-coko reviewed (approved):
Looks like far too many headers but not horrible 👍 
![](https://media4.giphy.com/media/SmoCFhZCi1kzu/giphy.gif)

markocikos commented:
Forwarded to https://github.com/liferay-commerce/liferay-portal/pull/1057
